spec/unit/functions/any_resources_of_type_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'spec_helper'
+require 'puppet/pops'
+require 'puppet/loaders'
+
+describe "the 'any_resources_of_type' function" do
+  after(:all) { Puppet::Pops::Loaders.clear }
+
+  # This loads the function once and makes it easy to call it
+  # It does not matter that it is not bound to the env used later since the function
+  # looks up everything via the scope that is given to it.
+  # The individual tests needs to have a fresh env/catalog set up
+  #
+  let(:loaders) { Puppet::Pops::Loaders.new(Puppet::Node::Environment.create(:testing, [File.expand_path(fixtures('modules'))])) }
+  let(:func) { loaders.private_environment_loader.load(:function, 'any_resources_of_type') }
+
+  before :each do
+    # A fresh environment is needed for each test since tests create resources
```

Comment:
i think this can be replaced using `let` https://www.betterspecs.org/#let

```ruby
  let(:environment) { Puppet::Node::Environment.create(:testing, [File.expand_path(fixtures('modules'))]) }
  let(:node) { Puppet::Node.new('yaynode', :environment => environment) }
  let(:compiler) { Puppet::Parser::Compiler.new(node) }
  let(:scope) { Puppet::Parser::Scope.new(compiler) }
```

Reply:
And update the test of the usage in examples

Reply:
Tried this, but got:

> `let` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block)

Reply:
`let` should be defined outside `before` blocks, that should work.

Reply:
hmm okay, didn't know `let` blocks refreshed before each test, but seems to be working :+1: